eventuate-messaging-kafka-common/src/main/java/io/eventuate/messaging/kafka/common/EventuateKafkaMultiMessageConverter.java

```diff
@@ -0,0 +1,91 @@
+package io.eventuate.messaging.kafka.common;
+
+import java.io.*;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+public class EventuateKafkaMultiMessageConverter {
+  public static final String MAGIC_ID = "a8c79db675e14c4cbf1eb77d0d6d0f00"; // generated UUID
+  public static final byte[] MAGIC_ID_BYTES = MAGIC_ID.getBytes();
+
+  public byte[] convertMessagesToBytes(List<EventuateKafkaMultiMessageKeyValue> messages) {
+    ByteArrayOutputStream binaryStream = new ByteArrayOutputStream();
+
+    try {
+      binaryStream.write(MAGIC_ID_BYTES);
+
+      for (EventuateKafkaMultiMessageKeyValue message : messages) {
+        byte[] keyBytes = Optional.ofNullable(message.getKey()).map(String::getBytes).orElse(new byte[0]);
```

Comment:
Shouldn't this use EventuateKafkaMultiMessageBuilder?

Reply:
Probably this method is not needed. As I wrote in email, there is some copy paste. I will need to make a clean up.